CMakeLists.txt

```diff
@@ -226,22 +248,26 @@ if (ARCH_IA32 OR ARCH_X86_64)
 endif()
 
 if (ARCH_AARCH64)
-    if (BUILD_SVE2_BITPERM)
+    if (BUILD_SVE2_BITPERM AND NOT SVE2_BITPERM_FOUND)
         set(GNUCC_ARCH "${GNUCC_ARCH}+sve2-bitperm")
-    elseif (BUILD_SVE2)
+    elseif (BUILD_SVE2 AND NOT SVE2_FOUND)
         set(GNUCC_ARCH "${GNUCC_ARCH}+sve2")
-    elseif (BUILD_SVE)
+    elseif (BUILD_SVE AND NOT SVE_FOUND)
```

Comment:
I may be misunderstanding it here, If BUILD_SVE is set, and SVE is NOT found, then enable it? Should it not be the other way around?

Reply:
you understood correctly, I'll explain the logic.  

since this section only adds the various sve flags. to not break the compilation, they will be added only if they don't exist currently in the arch field ${GNUCC_ARCH}.  
inside the arch folder there is a check if sve is actually supported on the system (this is in case one of the sve flags has been activated) and errors out if it doesn't (as per the original behavior).  
this way one of the SVE flags isn't turned on automatically but according to user request, and in that specific instance, it doesn't break the currently present arch field with an addition at the end of the line.

Reply:
It seems counter-intuitive. What if they exist in the GNUCC_ARCH but user explicitly sets BUILD_SVE=off? SVE Code paths will be disabled, but that does not mean the compiler cannot optimize the build using actual SVE instructions. Which may be confusing to the user, and it may lead to SIGILL on non-SVE systems. But I think that's the case also for AVX2/AVX512 on Intel. I'm afraid, the hardware features detection should be reworked entirely soon. In any case for this particular problem, could you please rephrase the logic so that it's more obvious why this is so?

Reply:
the current logic is 
if (not found and yes requested) 
    add flags
    
I'm not sure how to rephrase this as this is very specific condition and not hidden.

> but user explicitly sets BUILD_SVE=off? why would a user turn off something they didn't turn on? is there an example in the cmake for this logic somewhere?

Reply:
ok, what I mean is this: regardless if SVE found or not, it should be enabled only if BUILD_SVE is set to true, so the old logic was enough. In order to build for SVE, you only really need compiler support, not platform support (now running the tests is another matter ofc). But I now see what your method solves, you need the extra check so as not to add "+sve" a second time to the flags -which would break the build. Ok, this will work, but for the long term I will have to think of a better solution for all architectures, so you can expect this detection system to change in the future in general.